crates/ecs/src/lib.rs

```diff
@@ -810,6 +821,11 @@ impl World {
                 let target_archetype_idx =
                     self.move_entity_components_to_new_archetype(entity, source_type_ids)?;
 
+                let mut target_type_ids_vec: Vec<TypeId> = target_type_ids.into_iter().collect();
+                target_type_ids_vec.sort();
```

Comment:
You should leave a comment describing why these need to be sorted in order for it to work.